public/app/core/components/editors/DashboardPicker.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { FC, useCallback, useEffect, useState } from 'react';
+import debounce from 'debounce-promise';
+import { SelectableValue, StandardEditorProps } from '@grafana/data';
+import { DashboardSearchHit } from 'app/features/search/types';
+import { backendSrv } from 'app/core/services/backend_srv';
+import { AsyncSelect } from '@grafana/ui';
+
+export interface DashboardPickerOptions {
+  placeholder?: string;
+  isClearable?: boolean;
+}
+
+const getDashboards = (query = '') => {
+  return backendSrv.search({ type: 'dash-db', query }).then((result: DashboardSearchHit[]) => {
+    return result.map((item: DashboardSearchHit) => ({
+      uid: item.uid,
+      value: item.uid,
+      label: `${item?.folderTitle ?? 'General'}/${item.title}`,
+    }));
+  });
+};
+
+export const DashboardPicker: FC<StandardEditorProps<string, any, any>> = ({ value, onChange, item }) => {
+  const [current, setCurrent] = useState<SelectableValue<string>>();
+
+  useEffect(() => {
```

Comment:
Do we need this `useEffect` logic at all here? The current dashboard picker works without it.

Reply:
@Clarity-89 does it?  The one in playlist never has a selected element, and only responds to changes -- the async selector needs the `SelectableValue` and can not use the raw value

Reply:
At least the one in reporting does, we use it with our Form component, so maybe that's the difference.

Reply:
Ahh yes -- that has a SelectableValue<any> as a parent element and does not try to match it with the values in the async lookup.

See: https://github.com/grafana/grafana-enterprise/pull/1887

Reply:
Ah I see, still feels a bit weird doing this, as I think this is smth `AsyncSelect` should be doing instead, but seems like it can't match the values by ids internally. 

I checked and it looks like AsyncSelect loads all the dashboards on mount in any case, so I think we'd use the normal Select, load all of the dashboards on mount and then just match the selected one from there by uid, smth like this (not tested):
```
export const DashboardPicker: FC<StandardEditorProps<string>> = ({ value, onChange, item }) => {
  const [dashboards, setDashboards] = useState<SelectableValue[]>([]);
  const [current, setCurrent] = useState<SelectableValue | undefined>();

  useEffect(() => {
    (async function loadDashboards() {
      const res = await getDashboards();
      setDashboards(res);
    })();
  }, []);

  useEffect(() => {
    if (value) {
      setCurrent(dashboards.find((db) => db?.uid === value));
    } else {
      setCurrent(undefined);
    }
  }, [value, dashboards]);

  const onPicked = useCallback(
    (sel: SelectableValue<string>) => {
      onChange(sel.value);
    },
    [onChange]
  );

  const { placeholder, isClearable } = item?.settings ?? {};

  return (
    <Select
      menuShouldPortal
      isClearable={isClearable}
      options={dashboards}
      onChange={onPicked}
      placeholder={placeholder ?? 'Select dashboard'}
      noOptionsMessage="No dashboards found"
      value={current}
    />
  );
};
```
I've added the second `useEffect` to account for the need to clear the value, not sure if that's necessary.

Reply:
Think the cleanest way to get the currently selected dashboard(passed via value prop) is to use [`useAsync` hook](https://github.com/streamich/react-use/blob/master/docs/useAsync.md). 

I would opt for @Clarity-89 solution for loading all dashboards upfront, but there are cases when this is going to be frustrating:
- What if other user aded dashboard in the meantime?
- I'm not 100% sure we can get all the dashboards by API call. Think there is some cap on the number of dashboards returned, what may be quite problematic in large instances.

Reply:
Simple select would be fine... except the API only returns the first page, so :man_shrugging:

Reply:
I switched to useAsync